Upgrade to hyfetch v2.0.2

EtaoinWu opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Since `v2.0.0`, upstream has updated from a Python package to a Rust package with Python fallback. The upstream Python package fails to build in conda-forge's target version due to the dependency `fastfetch`, which [refuses](https://github.com/fastfetch-cli/fastfetch/issues/808) to be compatible with the glibc version available for EL7 and EL8.

After communicating with upstream @hykilpikonna, we decided to ship only the Rust package in this feedstock, as conda-forge has binary distribution capabilities.

### Maintenance

We have also switched to recipe v1 with `rattler-build`.

EtaoinWu committed:
upgrade to hyfetch v2.0.2; move to rust build

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipe/recipe.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `${{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17721560329](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17721560329). Examine the logs at this URL for more detail.</sub>

EtaoinWu committed:
add a dependency on stdlib("c") as required by linter.
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.14.19.31.04

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

EtaoinWu committed:
fix windows build problem by patching path handling

EtaoinWu committed:
whatever I'll just remote debug

EtaoinWu committed:
fix: window build

EtaoinWu committed:
fix: typo in command

EtaoinWu committed:
fix: environment variable in command for windows

EtaoinWu commented:
@conda-forge-admin, please rerender

EtaoinWu commented:
Finally good at this moment... Will push an issue to upstream to refactor their build.rs

Commit:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.09.27.05.44.0
